### PR TITLE
Move HandleContainer and Tensor Ops descriptions from burn-fusion to burn-tensor

### PR DESCRIPTION
**Pull request by syl20bnr**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

https://github.com/tracel-ai/burn/issues/1523

### Changes

Move `HandleContainer`  and Tensor operations descriptions to `burn-tensor` crate.
Removed the `FusionDevice` and replaced it with a `DeviceOps` trait bound to `Backend::Device`.

For now added modules to `burn-tensor` are excluded from `no-std` as they rely on `Arc`.

There are a few added docstring to check, I added the temporary prefix `--ADDED--` to them.

### CODE REVIEW

File: Cargo.toml
Comment (laggui) [inline Cargo.toml:143]: Are we keeping this profile or removing it? If removing, I'd tend to delete and not comment out personally.

**Review by laggui — COMMENTED** [review]
Big refactor 😄! Overall LGTM, will wait for @nathanielsimard's final approval since he is much more familiar with the affected changes than I am.

I think the added docstrings are fine. Let's make sure we don't forget to remove the `--ADDED--` prefix from those once accepted.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**
